src/jit/morph.cpp

```diff
@@ -14057,6 +14057,12 @@ GenTree* Compiler::fgMorphModToSubMulDiv(GenTreeOp* tree)
         denominator = fgMakeMultiUse(&tree->gtOp2);
     }
 
+    if (lvaLocalVarRefCounted)
```

Comment:
I think we need to do this for both the numerator and the denominator. For example, consider the following expression: `(mod (lclVar V0) (lclVar V1))`. After the transform, this becomes `(sub (lclVar V0) (mul (lclVar V1) (div (lclVar V0) (lclVar V1))))`, which adds an additional reference for both V0 and V1. Furthermore, I think that we can be more precise about when we do this, as we only need to bump ref counts for the numerator/denominator when they are lclVar nodes. We could even fold these checks into the `if` statements above, e.g.

```
    if (!numerator->OperIsLeaf())
    {
        numerator = fgMakeMultiUse(&tree->gtOp1);
    }
    else if (lvaLocalVarRefCounted && numerator->OperIsLocal())
    {
        lvaIncRefCnts(numerator);
    }
```

and similarly for the denominator.